worlds/yoshisisland/__init__.py

```diff
@@ -0,0 +1,380 @@
+        spoiler_handle.write(f"2-8: {self.level_name_list[15]}\n")
+
+        spoiler_handle.write(f"\n3-1: {self.level_name_list[16]}\n")
+        spoiler_handle.write(f"3-2: {self.level_name_list[17]}\n")
+        spoiler_handle.write(f"3-3: {self.level_name_list[18]}\n")
+        spoiler_handle.write(f"3-4: {self.level_name_list[19]}\n")
+        spoiler_handle.write(f"3-5: {self.level_name_list[20]}\n")
+        spoiler_handle.write(f"3-6: {self.level_name_list[21]}\n")
+        spoiler_handle.write(f"3-7: {self.level_name_list[22]}\n")
+        spoiler_handle.write(f"3-8: {self.level_name_list[23]}\n")
+
+        spoiler_handle.write(f"\n4-1: {self.level_name_list[24]}\n")
+        spoiler_handle.write(f"4-2: {self.level_name_list[25]}\n")
+        spoiler_handle.write(f"4-3: {self.level_name_list[26]}\n")
+        spoiler_handle.write(f"4-4: {self.level_name_list[27]}\n")
+        spoiler_handle.write(f"4-5: {self.level_name_list[28]}\n")
+        spoiler_handle.write(f"4-6: {self.level_name_list[29]}\n")
+        spoiler_handle.write(f"4-7: {self.level_name_list[30]}\n")
+        spoiler_handle.write(f"4-8: {self.level_name_list[31]}\n")
+
+        spoiler_handle.write(f"\n5-1: {self.level_name_list[32]}\n")
+        spoiler_handle.write(f"5-2: {self.level_name_list[33]}\n")
+        spoiler_handle.write(f"5-3: {self.level_name_list[34]}\n")
+        spoiler_handle.write(f"5-4: {self.level_name_list[35]}\n")
+        spoiler_handle.write(f"5-5: {self.level_name_list[36]}\n")
+        spoiler_handle.write(f"5-6: {self.level_name_list[37]}\n")
+        spoiler_handle.write(f"5-7: {self.level_name_list[38]}\n")
+        spoiler_handle.write(f"5-8: {self.level_name_list[39]}\n")
+
+        spoiler_handle.write(f"\n6-1: {self.level_name_list[40]}\n")
+        spoiler_handle.write(f"6-2: {self.level_name_list[41]}\n")
+        spoiler_handle.write(f"6-3: {self.level_name_list[42]}\n")
+        spoiler_handle.write(f"6-4: {self.level_name_list[43]}\n")
+        spoiler_handle.write(f"6-5: {self.level_name_list[44]}\n")
+        spoiler_handle.write(f"6-6: {self.level_name_list[45]}\n")
+        spoiler_handle.write(f"6-7: {self.level_name_list[46]}\n")
+        spoiler_handle.write("6-8: King Bowser's Castle")
+
+    def create_item(self, name: str) -> Item:
+        data = item_table[name]
+        classification = self.item_classifications[data.classification]
+
+        item = Item(name, classification, data.code, self.player)
+
+        return item
+
+    def create_regions(self) -> None:
+        init_areas(self, get_locations(self))
+
+    def get_filler_item_name(self) -> str:
+        trap_chance: int = self.options.trap_percent.value
+
+        if self.random.random() < (trap_chance / 100) and self.options.traps_enabled.value == 1:
+            return self.random.choice(trap_items)
+        else:
+            return self.random.choice(filler_items)
+
+    def set_rules(self) -> None:
+        if self.options.stage_logic == 0:
```

Comment:
I recommend using the actual option values in these conditions, instead of magic numbers. This way, if they ever change, the code will still work

Reply:
How so? I don't particularly forsee any of the actual numbers changing, and it feels somewhat unnecessary? At least to me.

Reply:
well, first, just readability. Reading this code doesn't say anything. "If stage logic is 0", "If stage logic is 1" is very obscure, as it is not clear what exactly are 0 and 1.
Usually, you'd try to make the code readable. Someone should be able to pop in with zero context, read a few lines, and know what happens there.

Reply:
For reference

```suggestion
        if self.options.stage_logic == self.options.stage_logic.option_strict:
```

You can also import the option itself and do

```suggestion
        if self.options.stage_logic == StageLogic.option_strict:
```

Pick your poison.

Reply:
i really don't like the idea of comparing an instance to a member of itself; it really doesn't make any sense. comparing it to the class attribute is very different.